ReactSkia/components/RSkComponentTextInput.cpp

```diff
@@ -89,7 +89,7 @@ void RSkComponentTextInput::drawTextInput(SkCanvas *canvas,
   paragraph_->paint(canvas, frame.origin.x + layout.contentInsets.left, frame.origin.y + layout.contentInsets.top + yOffset);
 
   // draw Cursor
-  if (!caretHidden_) {
+  if (!caretHidden_ && isInEditingMode_) {
```

Comment:
if possible, interchange the variable. if it is not in Editing mode no need to check caret Hidden right ?  
if (isInEditingMode_ && !caretHidden_ )